File: mobile/src/main/java/com/ourygo/ygomobile/util/AppInfoManagement.java
package com.ourygo.ygomobile.util;

import android.content.SharedPreferences;

import com.ourygo.ygomobile.OYApplication;

import cn.garymb.ygomobile.App;
import cn.garymb.ygomobile.utils.SystemUtils;

/**
 * Create By feihua  On 2022/8/21
 */
public class AppInfoManagement {
    private static final AppInfoManagement ourInstance = new AppInfoManagement();
    private boolean isNewVersion;

    private AppInfoManagement() {
        isNewVersion = false;
    }

    public static AppInfoManagement getInstance() {
        return ourInstance;
    }

    public boolean isNewVersion() {
        if (isNewVersion)
            return true;
        SharedPreferences sh = App.get().getSharedPreferences("AppVersion", App.get().MODE_PRIVATE);
        int vercode = SystemUtils.getVersion(OYApplication.get());
        int vn = sh.getInt("versionCode", 0);
        if (vn < vercode) {
            sh.edit().putInt("versionCode", vercode).apply();
            isNewVersion = true;
        } else {
            isNewVersion = false;
        }
        return isNewVersion;
    }

    public void setNewVersion(boolean newVersion) {
        isNewVersion = newVersion;
    }

    //软件关闭时一定要调用
    public void close() {
        isNewVersion = false;
    }
}

File: mobile/src/main/java/com/ourygo/ygomobile/util/AppInfoManagement.kt
package com.ourygo.ygomobile.util

import android.content.Context
import cn.garymb.ygomobile.App
import cn.garymb.ygomobile.utils.SystemUtils
import com.ourygo.ygomobile.OYApplication

/**
 * Create By feihua  On 2022/8/21
 */
object AppInfoManagement {
    private var isNewVersion = false
    fun isNewVersion(): Boolean {
        if (isNewVersion) return true
        val sh = App.get().getSharedPreferences("AppVersion", Context.MODE_PRIVATE)
        val vercode = SystemUtils.getVersion(OYApplication.get())
        val vn = sh.getInt("versionCode", 0)
        isNewVersion = if (vn < vercode) {
            sh.edit().putInt("versionCode", vercode).apply()
            true
        } else {
            false
        }
        return isNewVersion
    }

    fun setNewVersion(newVersion: Boolean) {
        isNewVersion = newVersion
    }

    //软件关闭时一定要调用
    fun close() {
        isNewVersion = false
    }
}